tools/pubsys/src/repo/fetch_variant/mod.rs

```diff
@@ -0,0 +1,263 @@
+    };
+
+    info!("Downloading targets to {tempdir_path:?}");
+    for target in target_names.clone() {
+        download_target(target).await?;
+    }
+
+    info!("Cleaning up {outdir:?}");
+    tokio::fs::remove_dir_all(outdir)
+        .await
+        .context(error::CleanDirSnafu { path: outdir })?;
+
+    info!("Moving targets to {outdir:?}");
+    tokio::fs::create_dir_all(outdir)
+        .await
+        .context(error::CreateDirSnafu { path: outdir })?;
+    for target in target_names {
+        let mut tmpdir_target_path = PathBuf::from(tempdir.path());
+        tmpdir_target_path.push(target.raw());
+        let mut outdir_target_path = outdir.clone();
+        outdir_target_path.push(target.raw());
+        tokio::fs::copy(tmpdir_target_path, outdir_target_path)
+            .await
+            .context(error::CopyTargetSnafu)?;
+    }
+
+    tempdir.close().context(error::CloseTempDirSnafu)?;
+    Ok(())
+}
+
+/// Fetch the variant and download its image targets
+async fn fetch_variant(
+    root_role_path: &PathBuf,
+    metadata_url: Url,
+    targets_url: &Url,
+    outdir: &PathBuf,
+    filename_prefix: &str,
+    variant_manifest: &PathBuf,
+    variant: &str,
+) -> Result<(), Error> {
+    // Load the repository
+    let repo = RepositoryLoader::new(
+        &repo::root_bytes(root_role_path).await?,
+        metadata_url.clone(),
+        targets_url.clone(),
+    )
+    .load()
+    .await
+    .context(repo_error::RepoLoadSnafu {
+        metadata_base_url: metadata_url.clone(),
+    })?;
+
+    let manifest_info = ManifestInfo::new(variant_manifest).context(error::ManifestParseSnafu)?;
+
+    let image_layout = manifest_info
+        .image_layout()
+        .context(error::MissingImageLayoutSnafu { variant })?;
+    let image_format = manifest_info.image_format();
+    let image_ext = match image_format {
+        Some(ImageFormat::Raw) | None => "img.lz4",
+        Some(ImageFormat::Qcow2) => "qcow2",
+        Some(ImageFormat::Vmdk) => "ova",
+    };
+
+    let targets = match image_format {
+        // Since the OVA will contain all of the necessary VMDKs, the partition plan is irrelevant.
+        Some(ImageFormat::Vmdk) => {
+            vec![format!("{filename_prefix}.{image_ext}")]
+        }
+        _ => match image_layout.partition_plan {
+            PartitionPlan::Split => {
+                vec![
+                    format!("{filename_prefix}.{image_ext}"),
+                    format!("{filename_prefix}-data.{image_ext}"),
+                ]
+            }
+            PartitionPlan::Unified => vec![format!("{filename_prefix}.{image_ext}")],
+        },
+    };
+    handle_download(&repo, outdir, &targets).await
+}
+
+/// Common entrypoint from main()
+pub(crate) async fn run(args: &Args, fetch_variant_args: &FetchVariantArgs) -> Result<(), Error> {
+    // If a lock file exists, use that, otherwise use Infra.toml
+    let infra_config = InfraConfig::from_path_or_lock(&args.infra_config_path, false)
```

Comment:
Hmm, Infra.lock isn't a thing anymore. But following the pattern is fine to be cleaned up later.